internal/suggestions/suggestion.go

```diff
@@ -22,42 +31,81 @@ func Suggest(docPath, maxSuggestions string) (string, error) {
 }
 
 func WriteSuggestions(g *git.Git, prNumber *int, out string) error {
-	body := parseOutput(out)
-	output := formatSuggestionsAndExplanations(body)
-	if len(output) > 0 {
-		// Writes suggestions and explanations in PR body
-		if err := g.WritePRBody(prNumber, output); err != nil {
+	commentsInfo, fileName := parseSuggestOutput(out)
+	body := formatBody(commentsInfo)
+	if body != "" {
+		// Writes suggestions and explanations with line number 0 PR body
+		if err := g.WritePRBody(prNumber, body); err != nil {
 			return fmt.Errorf("error writing PR body: %w", err)
 		}
 	}
 
+	for i := 0; i < len(commentsInfo.lineNums); i++ {
+		if commentsInfo.lineNums[i] != 0 {
+			comment := formatComment(commentsInfo.errs[i], commentsInfo.suggestions[i], commentsInfo.explanations[i], i+1)
+			if err := g.WritePRComment(prNumber, fileName, comment, commentsInfo.lineNums[i]); err != nil {
+				return fmt.Errorf("error writing PR comment: %w", err)
+			}
+		}
+	}
+
 	return nil
 }
 
-func formatSuggestionsAndExplanations(body prBodyInfo) string {
-	var output string
-	for i := 0; i < len(body.suggestions); i++ {
-		output += fmt.Sprintf("**Suggestion %d**: %s\n\n", i+1, body.suggestions[i])
-		output += fmt.Sprintf("**Explanation %d**: %s\n\n", i+1, body.explanations[i])
+func formatComment(err, suggestion, explanation string, index int) string {
+	var commentParts []string
+	commentParts = append(commentParts, fmt.Sprintf("**Error %d**: %s", index, err))
```

Comment:
Nice, I was noticing that it was a bit difficult to understand the PR bodies without the actual errors